great_expectations/data_context/migrator/cloud_migrator.py

```diff
@@ -143,81 +149,142 @@ def migrate_validation_result(
+        self._emit_warnings(
+            configuration_bundle=configuration_bundle, test_migrate=test_migrate
+        )
+        self._print_configuration_bundle_summary(
+            configuration_bundle=configuration_bundle
+        )
+
+        serialized_bundle = self._serialize_configuration_bundle(
+            configuration_bundle=configuration_bundle
         )
-        self._print_configuration_bundle(configuration_bundle)
+        serialized_validation_results = self._prepare_validation_results(
+            serialized_bundle=serialized_bundle
+        )
+
         if not test_migrate:
-            configuration_bundle_serializer = ConfigurationBundleJsonSerializer(
-                schema=ConfigurationBundleSchema()
-            )
-            configuration_bundle_response: AnyPayload = self._send_configuration_bundle(
-                configuration_bundle, configuration_bundle_serializer
+            self._send_configuration_bundle(serialized_bundle=serialized_bundle)
+            self._send_validation_results(
+                serialized_validation_results=serialized_validation_results,
             )
-            self._print_send_configuration_bundle_error(configuration_bundle_response)
-            self._break_for_send_configuration_bundle_error(
-                configuration_bundle_response
-            )
-        errors: List[
-            SendValidationResultsErrorDetails
-        ] = self._send_and_print_validation_results(test_migrate)
-        self._print_validation_result_error_summary(errors)
+
         self._print_migration_conclusion_message()
 
-    def _warn_if_test_migrate(self) -> None:
+    def _emit_warnings(
+        self, configuration_bundle: ConfigurationBundle, test_migrate: bool
+    ) -> None:
+        if test_migrate:
+            self._warn_about_test_migrate()
+        if not configuration_bundle.is_usage_stats_enabled():
+            self._warn_about_usage_stats_disabled()
+        if configuration_bundle.datasources:
+            self._warn_about_bundle_contains_datasources()
+
+    def _warn_about_test_migrate(self) -> None:
         pass
 
-    def _warn_if_usage_stats_disabled(self, is_usage_stats_enabled: bool) -> None:
+    def _warn_about_usage_stats_disabled(self) -> None:
         pass
 
-    def _build_configuration_bundle(self) -> ConfigurationBundle:
+    def _warn_about_bundle_contains_datasources(self) -> None:
         pass
 
-    def _print_configuration_bundle(
+    def _print_configuration_bundle_summary(
         self, configuration_bundle: ConfigurationBundle
     ) -> None:
         pass
 
-    def _send_configuration_bundle(
+    def _serialize_configuration_bundle(
+        self, configuration_bundle: ConfigurationBundle
+    ) -> dict:
+        serializer = ConfigurationBundleJsonSerializer(
+            schema=ConfigurationBundleSchema()
+        )
+        serialized_bundle = serializer.serialize(configuration_bundle)
+        return serialized_bundle
```

Comment:
Placed the serializer within the actual method since it never changes but lmk if you disagree!

Reply:
I think that is fine and simpler, these are all locally contained private methods so we can always move it out to a param if we need to inject a different serializer in the future without a breaking change.

Reply:
Can we just define a `.dict()` or `.to_dict()` method for the `ConfigurationBundle` class?

Reply:
@Kilo59 I thought we were trying to move away from objects defining their own serialization, especially since we need to serialize differently to yaml, json or dict e.g. for datasources (for storing in `great_expectations.yml`, sending to the cloud store, and printing to the screen in `context.list_datasources()`)